scrape.py

```diff
@@ -148,23 +148,24 @@ def extract_listings(soup):
     Returns a dict that maps listing IDs to listing details.
     """
     listings = {}
-    for div in soup.find_all('div', class_='resultlist_entry_data'):
-        for a in div.find_all('a'):
+    for entry in soup.find_all('article', class_="result-list-entry"):
+        for a in entry.find_all('a'):
             if a.get('href', '').startswith('/expose/'):
                 listing_id = a.get('href').split('/')[-1]
                 break
         else:
             # Couldn't find listing's ID
             continue
-        street_span = div.find('span', class_='street')
+        street_span = entry.find('div', class_='result-list-entry__address').find('span').contents[0]
         if not street_span:
             continue
         street, number, suburb = parse_address(unicode(street_span.string))
-        for dd in div.find_all('dd', class_='value'):
+        for dl in entry.find_all('dl', class_='result-list-entry__primary-criterion'):
+            dd = dl.find('dd')
             content = unicode(dd.string).strip()
-            if content.endswith('€'):
+            if content.endswith(' €'):
```

Comment:
The new check covers less cases then the old (it needs an additional space). Why is that necessary?